app/controllers/application_controller.rb

```diff
@@ -249,9 +311,21 @@ def authorize_global
   # Action can be:
   # * a parameter-like Hash (eg. { controller: '/projects', action: 'edit' })
   # * a permission Symbol (eg. :edit_project)
-  def do_authorize(action, global: false)
-    context = @project || @projects
-    is_authorized = User.current.allowed_to?(action, context, global:)
+  def do_authorize(action, global: false) # rubocop:disable Metrics/PerceivedComplexity
+    # OpenProject::Deprecation.deprecate_method(ApplicationController, :do_authorize)
+    # context = @project || @projects
+    # old_authorized = User.current.allowed_to?(action, context, global:)
+
+    is_authorized = begin
+      if global
+        User.current.allowed_based_on_permission_context?(action)
+      else
+        User.current.allowed_based_on_permission_context?(action, project: @project || @projects,
+                                                                  entity: @work_package || @work_packages)
+      end
+    rescue Authorization::UnknownPermissionError
+      false
+    end
```

Comment:
This would be the quick & dirty version before we switch everything over to the the proper `before_action`. I quite like the approach here. WDYT?